File: 0242-valid-anagram/0242-valid-anagram.java
class Solution {
    int CHAR = 256;
    public boolean isAnagram(String s, String t) {
        if (s.length() != t.length())
        {
            return false;
        }
        int count[] = new int[CHAR];
        for (int i = 0;i<s.length();i++)
        {
            count[s.charAt(i)]++;
            count[t.charAt(i)]--;
        }
        for (int i = 0;i<s.length();i++)
        {
            if (count[s.charAt(i)]!=0)
            {
                return false;
            }
        }
        return true;
    }
}

File: 0242-valid-anagram/0242-valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        shash = defaultdict(int)
        for i in s:
            shash[i] += 1
        for j in t:
            if j not in shash:
                return False
            shash[j] -= 1
            if shash[j] == 0:
                del shash[j]
        return len(shash) == 0